hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/RenameAtomicityUtils.java

```diff
@@ -249,6 +255,13 @@ public void preRename(List<BlobProperty> blobPropertyList) throws IOException {
     }
   }
 
+  private Throwable getWrappedException(final IOException e) {
+    if(e.getCause() != null) {
+      return e.getCause().getCause();
```

Comment:
Why do we need two calls to getCause() ?

Reply:
Actually, the exception gets wrapped twice:

first at: https://github.com/saxenapranav/hadoop/blob/305f57ef0a0b779d297c319e2d88587a45c64a3b/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java#L859

then, second at: https://github.com/saxenapranav/hadoop/blob/305f57ef0a0b779d297c319e2d88587a45c64a3b/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java#L658